be/src/io/shared_buffered_input_stream.cpp

```diff
@@ -170,4 +170,14 @@ StatusOr<std::string_view> SharedBufferedInputStream::peek(int64_t count) {
     return std::string_view((const char*)buf, count);
 }
 
+int64_t SharedBufferedInputStream::estimated_mem_usage() const {
+    int64_t mem_usage = 0;
+    for (const auto& [_, sb] : _map) {
+        mem_usage += sb.size;
+    }
+    // add 50% overhead
+    mem_usage += mem_usage / 2;
```

Comment:
What does the 50% overhead include? For the metadata structure, maybe it is more accurate to estimate it according the map node count.

Reply:
the estimation is not because of memory of map, it's because in parquet parser, for compressed page, we will allocate a buffer to decompress it. That's almost make a copy.